matplotcheck/raster.py

```diff
@@ -74,22 +75,18 @@ def _which_label(self, label, all_label_options):
         string that is the first entry in the internal list which label is
         matched with. If no match is found, return value is None
         """
-        for label_opts in all_label_options:
-            for s in label_opts:
-                if s in label:
-                    return label_opts[0]
+        for label_option in all_label_options:
```

Comment:
i definitely think we need more tests. 
let's have a fixture that as two legends for a plot and see how it works with the changes as is. i suspect that it could be problematic but let's test it out and then see what we need to do to accomodate legends that are more complex.  just create a plot with two legends somewhat like we do with geopandas objects and see how this works!!